File: LeetCode/python3/1.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in d:
                return [d[diff], i]
            d[nums[i]] = i
        return []

File: LeetCode/golang/1.go
func twoSum(nums []int, target int) []int {
    u := make(map[int]int)
    for i,value := range nums {
        diff := target - value
        val, ok := u[diff]
        if ok {
            return []int{val, i}
        }
        u[value] = i
    }
    return []int{}
}
